src/ngResource/resource.js

```diff
@@ -656,6 +663,14 @@ angular.module('ngResource', ['ng']).
             if (!isInstanceCall && cancellable) {
               timeoutDeferred = $q.defer();
               httpConfig.timeout = timeoutDeferred.promise;
+
+              if(simulatedTimeout) {
+                simulatedTimeoutPromise = $timeout(function() {
+                  if(timeoutDeferred) {
+                    timeoutDeferred.resolve();
+                  }
+                }, simulatedTimeout);
```

Comment:
You are right, timeoutDeferred.resolve can be evaluated instantly (and not inside the timeout handler).